process/block/alteredaccounts/alteredAccountsProvider.go

```diff
@@ -0,0 +1,348 @@
+	alteredAccounts := make(map[string]*indexer.AlteredAccount)
+	var err error
+	for address, markedAccount := range markedAccounts {
+		err = aap.processMarkedAccountData(address, markedAccount.tokens, alteredAccounts)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return alteredAccounts, nil
+}
+
+func (aap *alteredAccountsProvider) processMarkedAccountData(
+	encodedAddress string,
+	markedAccountTokens []*markedAlteredAccountToken,
+	alteredAccounts map[string]*indexer.AlteredAccount,
+) error {
+	addressBytes, err := aap.addressConverter.Decode(encodedAddress)
+	if err != nil {
+		log.Warn("cannot decode marked altered account address", "error", err)
+		return err
+	}
+
+	account, err := aap.accountsDB.LoadAccount(addressBytes)
+	if err != nil {
+		log.Warn("cannot load account when computing altered accounts",
+			"address", encodedAddress,
+			"error", err)
+		return err
+	}
+
+	userAccount, ok := account.(state.UserAccountHandler)
+	if !ok {
+		log.Warn("cannot cast AccountHandler to UserAccountHandler", "address", encodedAddress)
+		return err
+	}
+
+	alteredAccounts[encodedAddress] = &indexer.AlteredAccount{
+		Address: encodedAddress,
+		Balance: userAccount.GetBalance().String(),
+		Nonce:   userAccount.GetNonce(),
+	}
+
+	if len(markedAccountTokens) > 0 {
+		for _, tokenData := range markedAccountTokens {
+			err = aap.fetchTokensDataForMarkedAccount(encodedAddress, tokenData, alteredAccounts)
+			if err != nil {
+				log.Warn("cannot fetch token data for marked account", "error", err)
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func (aap *alteredAccountsProvider) fetchTokensDataForMarkedAccount(
+	encodedAddress string,
+	markedAccountToken *markedAlteredAccountToken,
+	alteredAccounts map[string]*indexer.AlteredAccount,
+) error {
+	nonce := markedAccountToken.nonce
+	nonceBigInt := big.NewInt(0).SetUint64(nonce)
+
+	tokenID := markedAccountToken.identifier
+
+	storageKey := []byte(core.ElrondProtectedKeyPrefix + core.ESDTKeyIdentifier)
+	storageKey = append(storageKey, []byte(tokenID)...)
+	if nonce > 0 {
+		storageKey = append(storageKey, nonceBigInt.Bytes()...)
+	}
+
+	address, err := aap.addressConverter.Decode(encodedAddress)
+	if err != nil {
+		log.Warn("cannot decode encoded address", "error", err)
+		return err
+	}
+
+	account, err := aap.accountsDB.LoadAccount(address)
+	if err != nil {
+		return err
+	}
+
+	userAccount, ok := account.(state.UserAccountHandler)
+	if !ok {
+		return nil
+	}
+
+	esdtTokenBytes, err := userAccount.RetrieveValueFromDataTrieTracker(storageKey)
+	if err != nil {
+		return err
+	}
+
+	var esdtToken esdt.ESDigitalToken
+	err = aap.marshalizer.Unmarshal(&esdtToken, esdtTokenBytes)
+	if err != nil {
+		return err
+	}
+
+	alteredAccount := alteredAccounts[encodedAddress]
+
+	if alteredAccount.Tokens == nil {
+		alteredAccount.Tokens = make([]*indexer.AccountTokenData, 0)
+	}
+
+	alteredAccount.Tokens = append(alteredAccount.Tokens, &indexer.AccountTokenData{
```

Comment:
Does this mean that if an account has, let's say 3 operations with `token1`, each of them with nonce = 1,2,3 => all of them will be indexed, right? Do we want that? Or do we only want to store the latest nonce operation?

Reply:
all tokens an address interacted with during the block will be indexed